File: 2020/src/bin/day01a.rs
fn main() {
    use std::io::{self, BufRead};

    let stdin = io::stdin();

    let numbers = stdin.lock().lines()
        .map(|expense| expense.unwrap().parse::<u32>().unwrap())
        .collect::<Vec<_>>();

    let target = 2020;

    let (v1, v2) = numbers
        .iter()
        .flat_map(|v1| numbers.iter().clone().map(move |v2| (v1, v2)))
        .find(|(&v1, &v2)| v1 + v2 == target).unwrap();


    println!("{}", v1*v2);
}

File: 2020/cpp/day01a.cpp
#include <iostream>
#include <vector>

int main() {
    int n;
    std::vector<int> numbers = std::vector<int>();
    while(std::cin >> n) {
        numbers.push_back(n);
    }

    int target = 2020;

    int res = -1;

    for(auto v1 : numbers) {
        for(auto v2: numbers) {
            if(v1 + v2 == target) {
                res = v1 * v2;
                break;
            }
        }
        if(res !=-1) break;
    }

    std::cout << res;
}
